File: backend/GaraCare.Domain/Entities/Customer.cs
namespace GaraCare.Domain.Entities;

public class Customer
{
    public int Id { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Address { get; set; }

    // Nullable — null nếu khách vãng lai (không có tài khoản đăng nhập).
    public int? UserId { get; set; }
    public User? User { get; set; }

    public ICollection<Vehicle> Vehicles { get; set; } = new List<Vehicle>();
    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
}

File: frontend/types/customer.ts
// Khớp 1-1 với DTO backend GaraCare.Application/DTOs/Customers.

export interface CreateCustomerRequest {
  fullName: string;
  phone: string;
  email: string;
  address?: string;
}

export interface CustomerResponse {
  id: number;
  fullName: string;
  phone: string | null;
  email: string | null;
  address: string | null;
  userId: number | null;
}
